From a97e3a3a35708609f1c0f22bb5cb9ecdb2998d3b Mon Sep 17 00:00:00 2001
From: Ankit Agarwal <ankit.agarwal@intel.com>
Date: Thu, 13 Jun 2024 09:04:39 +0530
Subject: [PATCH] Fixed Build errors.

Facing build errors due to older hal version and due to undeclared
apis using older c99 ISO formats.

Tracked-On: NA
Signed-off-by: Ankit Agarwal <ankit.agarwal@intel.com>
---
 health/aidl/Android.bp                                      | 4 ++--
 health/aidl/utils/libhealthloop/include/health/HealthLoop.h | 1 +
 sensors/aidl/libiio_client/xml.c                            | 1 +
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/health/aidl/Android.bp b/health/aidl/Android.bp
index 7c38c14..ab13d03 100644
--- a/health/aidl/Android.bp
+++ b/health/aidl/Android.bp
@@ -20,7 +20,7 @@ cc_defaults {
         "libcutils",
         "liblog",
         "libutils",
-        "android.hardware.health-V2-ndk",
+        "android.hardware.health-V3-ndk",
 
         // TODO(b/177269435): remove when BatteryMonitor works with AIDL HealthInfo.
         "libhidlbase",
@@ -39,7 +39,7 @@ cc_defaults {
     name: "libhealth_aidl_charger_defaults.intel",
     shared_libs: [
         // common
-        "android.hardware.health-V2-ndk",
+        "android.hardware.health-V3-ndk",
         "libbase",
         "libcutils",
         "liblog",
diff --git a/health/aidl/utils/libhealthloop/include/health/HealthLoop.h b/health/aidl/utils/libhealthloop/include/health/HealthLoop.h
index 8d89083..fdd38b6 100644
--- a/health/aidl/utils/libhealthloop/include/health/HealthLoop.h
+++ b/health/aidl/utils/libhealthloop/include/health/HealthLoop.h
@@ -15,6 +15,7 @@
  */
 #pragma once
 
+#include <functional>
 #include <memory>
 #include <mutex>
 #include <vector>
diff --git a/sensors/aidl/libiio_client/xml.c b/sensors/aidl/libiio_client/xml.c
index d2fe3b0..883dc15 100644
--- a/sensors/aidl/libiio_client/xml.c
+++ b/sensors/aidl/libiio_client/xml.c
@@ -21,6 +21,7 @@
 
 #include <errno.h>
 #include <libxml/tree.h>
+#include <libxml/parser.h>
 #include <string.h>
 
 static int add_attr_to_channel(struct iio_channel *chn, xmlNode *n)
-- 
2.45.2

